### ISSUE TITLE
[BUG] Failure to unpack steamrt-heavy (and maybe more prerequisites) on some distribution (Void), cause SteamVR to not be able to start

### ISSUE THREAD

**Issue by Naia-love** [ValveSoftware/SteamVR-for-Linux#511]
**Describe the bug**
SteamVR just don't even want so start for me, I click on start and after a few seconds it just "crash" (the button get back to play again)

**To Reproduce**
1. I just tried to launch SteamVR for the first time since I reinstalled my OS

**Expected behavior**
To run

**System Information (please complete the following information):**
- Distribution: Bedrock Linux (main strata is void, i have also arch and ubuntu stratum installed, i tried reinstallig steamvr and running it with them disabled btw it didn't change anything apparently)
- SteamVR version: Beta 1.22.8
- Steam client version: Mar 14 2022, at 19:49:17
- Opted into Steam client beta?: Nop, steamvr yes
- Graphics driver version: nvidia 510.60.02
- Gist for SteamVR System Information: I can't create it as I can't even get to the steamvr menu; here my system information from steam directly tho https://gist.github.com/Naia-love/830c2491d23d6d6b16000f51573f4d0e

**additional context**
As far I know I have all needed libs as i can run games (natives or proton one) without any problems.
the vrclient_vrstartup log file only have
```
pr 20 2022 20:58:34.665790 - ================================================================================================
Wed Apr 20 2022 20:58:34.665979 - ================================================================================================
Wed Apr 20 2022 20:58:34.666137 - vrstartup 1.22.8 startup with PID=21538, config=/home/naia/.steam/steam/config, runtime=/home/naia/.local/share/Steam/steamapps/common/SteamVR
Wed Apr 20 2022 20:58:34.666278 - vrclient type=VRApplication_Bootstrapper
```
No other log file exists (for some reason the vrmonitor logfile actually got created and have only one entry out of my idk 20+ tries):
```
mer. avril 20 2022 20:07:11.916221 - //==============================================================================================
mer. avril 20 2022 20:07:11.916435 - ================================================================================================
mer. avril 20 2022 20:07:11.916625 - ================================================================================================
mer. avril 20 2022 20:07:11.916863 - vrmonitor 1.22.8 startup with PID=13703, config=/home/naia/.steam/steam/config, runtime=/home/naia/.local/share/Steam/steamapps/common/SteamVR
mer. avril 20 2022 20:07:11.917126 - Tools Path: /home/naia/.local/share/Steam/steamapps/common/SteamVR/tools exists.
mer. avril 20 2022 20:07:11.917317 - Demo Path: /home/naia/.local/share/Steam/steamapps/common/SteamVR/demo not found.
mer. avril 20 2022 20:07:14.917846 - Default Playback Audio Devices: ,  (Comm)
mer. avril 20 2022 20:07:14.917959 - Default Record Audio Device: ,  (Comm)
mer. avril 20 2022 20:07:14.918066 - TerminateZombieProcesses()
mer. avril 20 2022 20:07:15.005785 - [System] Detected first setup.
mer. avril 20 2022 20:07:18.049867 - [System] Transition from 'SteamVRSystemState_Off' to 'SteamVRSystemState_Startup'.
mer. avril 20 2022 20:07:18.878333 - [System] Runtime: 1650333605 250820  STEAMVR
```
Also trying to launch steamvr directly with `~/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh` I learned that it create dump, here my /tmp/dumps folder (nb: it doesn't have dumps of all my tries as i restarted my pc meantime)
[dumps.tar.gz](https://github.com/ValveSoftware/SteamVR-for-Linux/files/8524673/dumps.tar.gz)

**Comment by okawo80085** [comment]
Did it show you a prompt to finish SteamVR setup on first launch?

**Comment by Naia-love** [comment]
> Did it show you a prompt to finish SteamVR setup on first launch?
> 
> 

yes First launch after installing it did showed the prompt and then asked for my sudo passwd, which i did. And then nothing happened just "crashed", after reinstalling it it asked again but yea nothing

**Comment by okawo80085** [comment]
And just to be clear, it keeps asking you to finish setup on every launch now or?

**Comment by Naia-love** [comment]
> And just to be clear, it keeps asking you to finish setup on every launch now or?

oh no sorry i miss phrased, it just asked me once after first launch after install and then only once after first launch after a reinstall

**Comment by Naia-love** [comment]
> Oh gotcha, can you run `tree ~/.steam/root/steamapps/common/SteamVR` and post the output?

[Here you go](https://github.com/ValveSoftware/SteamVR-for-Linux/files/8534264/tree.txt)

**Comment by okawo80085** [comment]
You're missing almost a thousand files compared to my Linux SteamVR beta install, verify the integrity of the tool files in Steam

**Comment by Naia-love** [comment]
> You're missing almost a thousand files compared to my Linux SteamVR beta install, verify the integrity of the tool files in Steam
> 
> [tree diff](https://gist.github.com/okawo80085/53cb52ced48f61c7e2380371c2c96bc1)

I already did that more than once, and it's just say "all 5129 files successfully validated"

**Comment by okawo80085** [comment]
Thats... unusual to say the least, anything interesting in the Steam console when you start it? It should usually say the reason for the crash there

**Comment by Naia-love** [comment]
> Thats... unusual to say the least, anything interesting in the Steam console when you start it? It should usually say the reason for the crash there

```
➜  ~ ~/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh
+++ dirname /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/
++ pwd
+ VRBINDIR=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin
+ [[ -n '' ]]
+ exec /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh
++ readlink -f /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
+ export STEAMVR_VRENV=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
+ STEAMVR_VRENV=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
+++ dirname /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/..
++ pwd
+ export STEAMVR_TOOLSDIR=/home/naia/.local/share/Steam/steamapps/common/SteamVR
+ STEAMVR_TOOLSDIR=/home/naia/.local/share/Steam/steamapps/common/SteamVR
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/../runtime
/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh: ligne 18 : cd: /home/naia/.local/share/Steam/steamapps/common/SteamVR/../runtime: Aucun fichier ou dossier de ce type
++ pwd
+ RUNTIMEDIR=/home/naia
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/../sdk
/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh: ligne 19 : cd: /home/naia/.local/share/Steam/steamapps/common/SteamVR/../sdk: Aucun fichier ou dossier de ce type
++ pwd
+ SDKDIR=/home/naia
+ case $(uname) in
++ uname
+ '[' -z '' ']'
+ exec /home/naia/.steam/root/ubuntu12_32/steam-runtime/run.sh /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh
++ readlink -f /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
+ export STEAMVR_VRENV=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
+ STEAMVR_VRENV=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
+++ dirname /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/..
++ pwd
+ export STEAMVR_TOOLSDIR=/home/naia/.local/share/Steam/steamapps/common/SteamVR
+ STEAMVR_TOOLSDIR=/home/naia/.local/share/Steam/steamapps/common/SteamVR
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/../runtime
/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh: ligne 18 : cd: /home/naia/.local/share/Steam/steamapps/common/SteamVR/../runtime: Aucun fichier ou dossier de ce type
++ pwd
+ RUNTIMEDIR=/home/naia
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/../sdk
/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh: ligne 19 : cd: /home/naia/.local/share/Steam/steamapps/common/SteamVR/../sdk: Aucun fichier ou dossier de ce type
++ pwd
+ SDKDIR=/home/naia
+ case $(uname) in
++ uname
+ '[' -z /home/naia/.steam/root/ubuntu12_32/steam-runtime ']'
+ VRSTARTUP=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrstartup
+ QT_DIR=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/qt
++ pwd
+ export LD_LIBRARY_PATH=/home/naia:/home/naia/bin/linux64:/home/naia/bin/linux64:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/qt/lib:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_32:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_64:/usr/lib32:/usr/lib/pipewire-0.3/jack:/usr/local/lib:/usr/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib
+ LD_LIBRARY_PATH=/home/naia:/home/naia/bin/linux64:/home/naia/bin/linux64:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/qt/lib:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_32:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_64:/usr/lib32:/usr/lib/pipewire-0.3/jack:/usr/local/lib:/usr/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib
+ export VRCOMPOSITOR_LD_LIBRARY_PATH=/home/naia:/home/naia/bin/linux64:/home/naia/bin/linux64:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/qt/lib:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_32:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_64:/usr/lib32:/usr/lib/pipewire-0.3/jack:/usr/local/lib:/usr/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib
+ VRCOMPOSITOR_LD_LIBRARY_PATH=/home/naia:/home/naia/bin/linux64:/home/naia/bin/linux64:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/qt/lib:/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_32:/home/naia/.steam/root/ubuntu12_32/steam-runtime/pinned_libs_64:/usr/lib32:/usr/lib/pipewire-0.3/jack:/usr/local/lib:/usr/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu:/home/naia/.steam/root/ubuntu12_32/steam-runtime/lib:/home/naia/.steam/root/ubuntu12_32/steam-runtime/usr/lib
+ exec /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh
+++ dirname /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh
++ cd /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/
++ pwd
+ VRBINDIR=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin
+ [[ -n /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/vrenv.sh ]]
+ SETUP_LOG=/tmp/SteamVRLauncherSetup.log
+ '[' -z /usr/bin/zenity ']'
+ case $(uname) in
++ uname
+ VRSTARTUP=/home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrstartup
+ SteamVRLauncherSetup
++ command -v getcap
+ '[' -x /usr/bin/getcap ']'
++ command -v setcap
+ '[' -x /usr/bin/setcap ']'
++ getcap /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrcompositor-launcher
+ [[ /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrcompositor-launcher cap_sys_nice=eip == *\c\a\p\_\s\y\s\_\n\i\c\e* ]]
+ return 0
+ '[' 0 '!=' 0 ']'
+ exec /home/naia/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrstartup
Using breakpad crash handler
Setting breakpad minidump AppID = 250820
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
crash_20220421214743_2.dmp[3499]: Uploading dump (out-of-process)
/tmp/dumps/crash_20220421214743_2.dmp
[1]    3393 segmentation fault  ~/.local/share/Steam/steamapps/common/SteamVR/bin/vrstartup.sh
➜  ~ crash_20220421214743_2.dmp[3499]: Finished uploading minidump (out-of-process): success = yes
crash_20220421214743_2.dmp[3499]: response: CrashID=bp-ac3a8ef8-7354-430c-96b2-590f12220421
crash_20220421214743_2.dmp[3499]: file ''/tmp/dumps/crash_20220421214743_2.dmp'', upload yes: ''CrashID=bp-ac3a8ef8-7354-430c-96b2-590f12220421''
```

Nothing interesting but so the dump, that i already shared here in my issue

**Comment by okawo80085** [comment]
Strange, i'll check out the dumps tomorrow, but if it fails on startup it's probably missing something/something is setup incorrectly

**Comment by kedodrill** [comment]
This is happening to me as well, I just installed SteamVR today. Looks like the sdk and runtime dirs in the SteamVR folder don't exist. @okawo80085 do you happen to have those dirs in your SteamVR folder? I'm not seeing the dirs in your diff, which I think would show up?

**Comment by okawo80085** [comment]
> This is happening to me as well, I just installed SteamVR today. Looks like the sdk and runtime dirs in the SteamVR folder don't exist. @okawo80085 do you happen to have those dirs in your SteamVR folder? I'm not seeing the dirs in your diff, which I think would show up?
> 
> If it's worth anything, I had to manually run setcap on vrcompositor-launcher in order to get past the initial setup popup.

Im pretty sure I have them, so I'll follow up on that tomorrow

**Comment by Naia-love** [comment]
> If it's worth anything, I had to manually run setcap on vrcompositor-launcher in order to get past the initial setup popup.
 oh? how do you to this?

**Comment by kedodrill** [comment]
```
2022-04-22T02:48:39.13311 kern.info: [ 1460.093551] vrstartup[15224]: segfault at 0 ip 00007f2297edc5ed sp 00007ffd072ed230 error 4 in vrclient.so[7f22979e2000+6cb000]
2022-04-22T02:48:39.13316 kern.info: [ 1460.093569] Code: fd ff 48 89 e9 4c 89 c2 48 8d 35 a3 f3 03 00 31 c0 48 8d 3d 9d 7a 04 00 e8 30 25 fd ff 41 55 41 54 49 89 f4 55 53 48 83 ec 08 <4c> 8b 2f 48 89 f7 e8 28 f3 b8 ff 49 8b 5d e8 48 89 c5 48 39 d8 48
```

[dump.tar.gz](https://github.com/ValveSoftware/SteamVR-for-Linux/files/8536472/dump.tar.gz)

https://gist.github.com/kedodrill/0a10efe0a6213240b3376d5cc38bb716

**Comment by okawo80085** [comment]
@kedodrill Yeah `steam-runtime-heavy` is present for my install of SteamVR, [full tree of my installed version](https://github.com/ValveSoftware/SteamVR-for-Linux/files/8541001/steamvr_tree_test2.txt)

Altho it's strange, i remember a similar bug when SteamVR just failed to load `steam-runtime-heavy`, but it completely missing from the install is new, is it missing in non beta SteamVR for you as well?

**Comment by Naia-love** [comment]
> @kedodrill Yeah `steam-runtime-heavy` is present for my install of SteamVR, [full tree of my installed version](https://github.com/ValveSoftware/SteamVR-for-Linux/files/8541001/steamvr_tree_test2.txt)
> 
> Altho it's strange, i remember a similar bug when SteamVR just failed to load `steam-runtime-heavy`, but it completely missing from the install is new, is it missing in non beta SteamVR for you as well?

Actually, I switched to beta because it wasn't working on stable for me, and so I wanted to try if beta was actually working or not ^^'

**Comment by Naia-love** [comment]
I tried unpacking the steam-runtime-heavy.tar.xz, I now seems to have all the files you have in your own steam-runtime-heavy folder? but I still get exactly the same error
[tree.txt](https://github.com/ValveSoftware/SteamVR-for-Linux/files/8541166/tree.txt)

**Comment by okawo80085** [comment]
It still fails to find the runtime, but the more pressing issue is that it should've been there.

@kisak-valve any idea on what went wrong during the setup or how to fix it?

**Comment by Naia-love** [comment]
any update on that?

**Comment by okawo80085** [comment]
I don't see any mentions of fixes in the recent SteamVR beta updates, im assuming it's still not working on your end as well

**Comment by Naia-love** [comment]
> I don't see any mentions of fixes in the recent SteamVR beta updates, im assuming it's still not working on your end as well

Yep, tried to update to last beta, uninstall and reinstall, but nothing for now ^^'

**Comment by okawo80085** [comment]
There has been a few Linux related updates, is your issue still present?

**Comment by okawo80085** [comment]
> I'm guessing this is related, I could try installing it again with like steam running in a terminal to try and check if errors appears here

Please do.

**Comment by Naia-love** [comment]
> > I'm guessing this is related, I could try installing it again with like steam running in a terminal to try and check if errors appears here
> 
> Please do just that.

for some reason it didn't crashed this time :^) 
and so nothing interesting in steam log when installing it

**Comment by DeepwaterCreations** [comment]
Another Void user here having this exact same issue. Same error, same missing files, same behavior, so on and so forth.
Regarding Steam crashing after install, it's been crashing for me too, but sort of randomly a handful of seconds after I first start it up. After restarting two or three times, it seems to stabilize.
I'll try to get a log from it when it isn't 2am.

**Comment by DeepwaterCreations** [comment]
...so of course, once I've said I'll try to get a log from it, it stops happening. Go figure.

**Comment by Orthestral** [comment]
@kedodrill I have the exact same segfault in dmesg, down to the same code

**Comment by optlink** [comment]
@Orthestral @kedodrill I also have the exact same message on Void Linux.

**Comment by DeepwaterCreations** [comment]
> ...so of course, once I've said I'll try to get a log from it, it stops happening. Go figure.

Yeah, it hasn't happened once since then. No idea what changed.
Of course, SteamVR is still broken, so maybe those two things were never related?

**Comment by Orthestral** [comment]
Steam also has around a 50% chance to crash a few moments after starting up, giving me this error in dmesg:
```
[  132.954473] CIPCServer::Thr[9437]: segfault at 0 ip 00000000cdbec396 sp 00000000eaffd764 error 4 in vrclient.so[cd76e000+65c000]
[  132.954485] Code: fc ff 57 ff 74 24 34 eb e6 8d 74 26 00 8d bc 27 00 00 00 00 55 57 56 53 e8 f0 f0 be ff 81 c3 6b 2c 1f 00 83 ec 18 8b 44 24 2c <8b> 28 8b 7d f4 ff 74 24 30 e8 6c 20 bb ff 83 c4 10 89 c6 39 f8 89
```

**Comment by mlugg** [comment]
Is there a workaround for this issue? I've been playing around with using an open-source stack to replace SteamVR (OpenComposite + Monado + libsurvive), which works-ish for some games, but it'd be nice to be able to play Alyx with my shiny new Index controllers.
It's quite frustrating that this seems to be a Steam client issue, since there's literally no way to just revert SteamVR to a version where it worked.

**Comment by okawo80085** [comment]
> since there's literally no way to just revert SteamVR to a version where it worked.

There might be, try to play around with beta versions of SteamVR, it has a few old releases for Linux, but those are like *old* old, somewhere around 1.15.x iirc

**Comment by DeepwaterCreations** [comment]
> > ...so of course, once I've said I'll try to get a log from it, it stops happening. Go figure.
> 
> Yeah, it hasn't happened once since then. No idea what changed. Of course, SteamVR is still broken, so maybe those two things were never related?

Finally caught it in action: https://pastebin.com/jpxLkezz
That's the console output after the second time it crashed in a row. Doesn't seem terribly illuminating to me, but there it is anyway.

If there's a particular file in steam/steam/logs or something that might be useful, someone let me know and I'll post it too.

**Comment by okawo80085** [comment]
From steam/logs:
* `vrserver.txt` - has logs from drivers and system
* `vrclient_*.txt` - different client logs
* `vrmonitor.txt` - the SteamVR window that pops up
* `vrcompositor.txt` - the compositor logs
* `vrdashboard.txt` - the dashboard logs
* `vrhw_*.txt` - hardware related logs

`vrserver.txt` is definitely worth looking into, because it should have the logs if SteamVR errors out on startup

**Comment by DeepwaterCreations** [comment]
Hmm... a lot of these logs, including vrserver.txt, don't have anything in them past last March 11. I guess that's the last time I used it before it broke or something. (Yikes.)  Here's vrserver.txt, for instance: 
[vrserver.txt](https://github.com/ValveSoftware/SteamVR-for-Linux/files/9732317/vrserver.txt)

vrclient_steam.txt seems to be looking for oculus drivers, despite the fact that I don't have an oculus. (Maybe that's normal?) Then post March 20, all the entries are just sort of empty?
[vrclient_steam.txt](https://github.com/ValveSoftware/SteamVR-for-Linux/files/9732240/vrclient_steam.txt) All of the log files that have entries more recent than March 11 are like that.

vrhw_*.txt don't even exist. There's only a couple of .vdf files.

**Comment by DeepwaterCreations** [comment]
Not sure if this has anything new in it, but I tried running Steam with `$ LIBGL_DEBUG=verbose steam` and copied all the console output that happened after I hit the "Launch" button for SteamVR:

https://pastebin.com/q3xDh8g3

Prior to that, before I've done anything, it does say:
`Refusing to init SteamVR build 9645429 because it crashed. Last successful version was 9645429`
Not sure if that's helpful - I already knew it was crashing. But maybe there's some init step that's being skipped, and it keeps being skipped because it keeps being skipped...?

**Comment by Orthestral** [comment]
I just tried it today, and it seems to work now! Will update if anything goes wrong.

**Comment by Naia-love** [comment]
For me seems that it got fixed! I read here and there that valve pushed a lot of linux fix recently, well seems it run now for me on void on last public version, with ALVR, was able to launch games too (well only hav vrchat installed rn, which dont work cuz of Void specific bug with eac uu sadness)
I dont think i even had to uninstall and install it back?
I just want to hear back from y'all before closing this issue ^^
~~when deckard valve?~~

**Comment by DeepwaterCreations** [comment]
Matches my experience: It's been working for me, too.
(Save for some sound problems, but that's clearly a separate issue.)
Thanks, Valve!

**Comment by okawo80085** [comment]
Ok so i was not able to reply to this for a while cuz of literal war, but i can now and i got a void distro now with SteamVR on it, the catch is that for it to work properly (or at all with nvidia drivers) the Steam package needs to be installed properly, its often missed by users when just installing using `xbps-install` and not looking at the output, but specifically this package has post install steps that require you to install some extra packages, it will likely stay that way because those packages are in different void repositories (its `void-repo-nonfree` and `void-repo-multilib-nonfree` iirc).
In my case those packages were the following (minus the nvidia specific packages)
```
libgcc-32bit libstdc++-32bit libdrm-32bit libglvnd-32bit
```
Without them either Steam wont start or SteamVR wont be able to complete initial setup -> wont be able to start either.

I've tested my particular build of void for a few months now, SteamVR is more or less stable, i have tracking, all of my devices are working, apps that launch under proton work fine as well, there is some tracking jittering, but im pretty sure thats cuz the nvidia curse broke motion smoothing for me (not sure why exactly, dunno if i'll get the time to debug it).
So all in all its working fine xD

**Comment by Naia-love** [comment]
> Ok so i was not able to reply to this for a while cuz of literal war, but i can now and i got a void distro now with SteamVR on it, the catch is that for it to work properly (or at all with nvidia drivers) the Steam package needs to be installed properly, its often missed by users when just installing using `xbps-install` and not looking at the output, but specifically this package has post install steps that require you to install some extra packages, it will likely stay that way because those packages are in different void repositories (its `void-repo-nonfree` and `void-repo-multilib-nonfree` iirc).
> In my case those packages were the following (minus the nvidia specific packages)
> ```
> libgcc-32bit libstdc++-32bit libdrm-32bit libglvnd-32bit
> ```
> Without them either Steam wont start or SteamVR wont be able to complete initial setup -> wont be able to start either.
> Its explained better in this void package files readme, also not blaming users for missing this thing, digging through void package templates is not something they are expected to do
> https://github.com/void-linux/void-packages/tree/master/srcpkgs/steam/files


while what you're talking is true,
 and indced without following these instruction steamvr (and actually a lot of games in steam, especialy with steam runtime or what) dont work

this issue was about an actual bug preventing the install, despite all the prerequisite correctly installed, and that was fixed along with the big steamvr linux patch around 1.27 (i guess, dont know exact version fixed)

**Comment by kisak-valve** [comment]
Closing per the last several comments.
